src/makeInfinityAdder.js

```diff
@@ -5,8 +5,26 @@
  * @return {function}
  */
 
-function makeInfinityAdder() {
-  // write code here
+function makeInfinityAdder(num1) {
+  let sum = num1 || 0;
+
+  const foo = (num2) => {
+    if (num2 === undefined) {
+      const result = sum;
+
+      sum = 0;
+
+      return result;
+    }
+
+    sum += num2;
+
+    return foo;
+  };
+
+  foo.valueOf = () => sum;
```

Comment:
You were used valueOf, offbeat solution. Nace.

Reply:
it's good that you reminded me, it's not needed here )